README.md

```diff
@@ -3,201 +3,180 @@
+  console.log(feeds)
+  // [
+  //   { feedpurpose, subfeed, keys }
+  // ]
+})
 ```
 
-#### addNewFeed(metafeedKeys, previous, feedpurpose, seed, feedformat, metadata)
-
-Generate a message to be posted on meta feed linking feed to a meta
-feed. Similar to `deriveFeedKeyFromSeed`, `feedformat` can be either
-`bendy butt` for a meta feed or `classic`. `metatada` is an optional
-dict.
-
 ```js
-const msg = sbot.metafeeds.messages.addNewFeed(metafeedKeys, null, 'main', seed, 'classic')
+sbot.metafeeds.find(metafeed, f => f.feedpurpose === 'mygame', (err, feed) => {
+  console.log(feed)
+  // { feedpurpose, subfeed, keys }
+})
 ```
 
-#### tombstoneFeed(metafeedKeys, previous, feedKeys, reason, cb)
+Notice that each `f` (and the result) is an object that describes the subfeed, with the fields:
 
-Generate a message to be posted on meta feed tombstoning a feed on a
-meta feed. `Previous` is the previous message on the meta feed in key
-value form.
+* `feedpurpose`: same string as used to create the subfeed
+* `subfeed`: the SSB identifier for this subfeed
+* `keys`: the [ssb-keys] compatible `{ curve, public, private, id }` object
+
+Finally, there are many cases where you want to create a subfeed **only if** it doesn't yet exist. For those purposes, use `findOrCreate`, which is a mix of `find` and `create`. It literally will internally call `find`, and only call `create` if `find` did not find a subfeed. For instance:
 
 ```js
-const previous = {
-  key: '%vv/XLo8lYgFjX9sM44I5F6la2FAp6iREuZ0AVJFp0pU=.bbmsg-v1',
-  value: {
-    previous: '%jv9hs2es5Pkw85vSOmLvzQh4HtosbCrVjhT+fR6GPr4=.bbmsg-v1',
-    ...
+const details = {
+  feedpurpose: 'mygame',
+  feedformat: 'classic',
+  metadata: {
+    score: 0,
+    whateverElse: true
   }
 }
 
-sbot.metafeeds.messages.tombstoneFeed(metafeedKeys, previous, mainKeys, 'No longer used', (err, tombstoneMsg) => {
-  sbot.db.publishAs(mfKey, tombstoneMsg, (err) => {
-    console.log("main is now tombstoned on meta feed")
-  })
-})
+sbot.metafeeds.findOrCreate(
+  metafeed,
+  f => f.feedpurpose === 'mygame',
+  details, // only used if the "find" part fails
+  (err, feed) => {
+    console.log(feed)
+  }
+)
 ```
 
-#### generateAnnounceMsg(metafeedKeys, cb)
+## API
 
-Generate the content of a message to be published on a main feed
-linking it to a meta feed.
+### `sbot.metafeeds.filter(metafeed, visit, cb)`
 
-```js
-sbot.metafeeds.messages.generateAnnounceMsg(metafeedKeys, (err, announceMsg) => {
-  sbot.db.publish(announceMsg, (err) => {
-    console.log("main feed is now linked to meta feed")
-  })
-})
-```
+*Looks for all subfeeds of `metafeed` that satisfy the condition in `visit`.*
 
-#### generateSeedSaveMsg(metafeedId, mainfeedId, seed)
+`metafeed` can be either `null` or a meta feed object `{ seed, keys }` (as returned by `create()`). If it's null, then the result will be an array containing one item, the root meta feed, or zero items if the root meta feed does not exist.
```

Comment:
I can see how this null would be useful in the getOrCreate case. I was thinking at first that null would allow one to search in all metafeeds, not only in your own.

Reply:
Oh, that's a feature that didn't even cross my mind when working on this module. Do you see use cases for that?

I think we're going to interact with other people's meta feeds in different modules. Like the replication-scheduler, or whatever "ssb-friends for meta feeds" alternative we come up, that's where I would expect to want to get other people's meta feeds. I imagined that ssb-meta-feeds is kind of like ssb-identities, handling only your own stuff, where `findOrCreate` is the core use case, and these `find`/`filter` alternatives are there just in case the developer knows what they're doing (e.g. if they want to check if the subfeed exists without creating it in case it doesn't).

Reply:
I was thinking of the case where if we do indexes of other feeds, then you might have a rule that there should be at least 3 of each indexes as one of the index writers might be offline for an extended period of time. In this case it would be good to be able to query for non-tombstoned indexes in all meta feeds you know of. But I think we should tackle that use case when / if we run into it.

Reply:
Yeah, this API seems flexible to allow that in the future